File: backend-dotnet/Domain/Entities/Tag.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities
{
    public class Tag
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        // Navigation: one-to-many
        public ICollection<PostTag> PostTags { get; set; } = [];
    }
}

File: backend-java/src/main/java/com/jerrygram/domain/entities/Tag.java
package com.jerrygram.domain.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "\"Tags\"")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Tag {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "\"Id\"")
    private UUID id;
    
    @Column(name = "\"Name\"", nullable = false, unique = true, length = 50)
    private String name;
    
    // Navigation: one-to-many
    @OneToMany(mappedBy = "tag", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @Builder.Default
    private List<PostTag> postTags = new ArrayList<>();
}